File: solutions/2025-06-19/solution.cpp
//Leetcode 2294 - Partition Array Such That Maximum Difference Is K
//https://leetcode.com/problems/partition-array-such-that-maximum-difference-is-k/description/?envType=daily-question&envId=2025-06-19

class Solution {
public:
    int partitionArray(vector<int>& nums, int k) {
        // initialization
        int count = 0;
        int n = nums.size();
        int i = 0;
        // sorting
        sort(nums.begin(), nums.end());
        // traversing the array
        while (i < n) {
            // start
            int start = nums[i];
            i++;
            // finding end

            while (i < n && nums[i] - start <= k) {
                i++;
            }
            count++;
        }
        return count;
    }
}; 

File: solutions/2025-06-19/solution.py
#Leetcode 2294 - Partition Array Such That Maximum Difference Is K
#https://leetcode.com/problems/partition-array-such-that-maximum-difference-is-k/description/?envType=daily-question&envId=2025-06-19

class Solution:
    def partitionArray(self, nums: List[int], k: int) -> int:
        #initialization
        count = 0
        n = len(nums)
        i = 0
        #sort
        nums.sort()
        #traversing
        while i<n:
            start = nums[i]
            i+=1
            while i<n and nums[i]-start <= k:
                i+=1
            count+=1
        return count